### ISSUE TITLE
Inconsistency between get template and get mapping

### ISSUE THREAD

**Issue by konradkonrad** [elastic/elasticsearch#14160]
I was trying to programmatically test, that my template creating code is doing the right thing. Unfortunately I failed, since the get template API and the get mappings API don't agree on the formatting of `dynamic: false`.

The get-mappings returns false as a string (`"false"`), whilst get-template returns a boolean.
```
curl -XPUT localhost:9200/_template/1 -d '{
  "template": "test*",
  "mappings": {
    "test": {
      "dynamic": false,
      "properties": {"field1": {"type": "string"}
      }
    }
  }
}'

curl -XPUT localhost:9200/test/test/1 -d '{"field1": "foo"}'
```
now the template looks like this:
```
curl "localhost:9200/_template/1?pretty"
{
  "1" : {
    "order" : 0,
    "template" : "test*",
    "settings" : { },
    "mappings" : {
      "test" : {
        "dynamic" : false,
        "properties" : {
          "field1" : {
            "type" : "string"
          }
        }
      }
    },
    "aliases" : { }
  }
}
```
and the mapping like this :(
```
curl "localhost:9200/test/_mappings?pretty"
{
  "test" : {
    "mappings" : {
      "test" : {
        "dynamic" : "false",
        "properties" : {
          "field1" : {
            "type" : "string"
          }
        }
      }
    }
  }
}
```
I found this on "1.6.2" -- feel free to dismiss if it was already addressed.

I can see, where this is coming from: since `dynamic` is not really boolean (because of `strict`), it needs to be realized as a string in the mapping. But couldn't the template API apply the same logic, so the `"mappings": {...` blocks in template and index-mappings become comparable?

Note: I "solved" the problem by creating the template with the string `"false"`, but for a user this is a little nagging inconsistency...

**Comment by clintongormley** [comment]
Hi @konradkonrad 

The mapping response is generated from actual Java objects. The template, however, is basically just a string that gets parsed only when an index is created.  Currently  you can put what you like in there and the first time it will throw an exception is when you try to create an index with it. 

Even after #2415 is implemented, I'm not sure whether we'll store the original string or the generated JSON.  Either way your current solution will continue to work.

Closing in favour of #2415
